@AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_routeCycle.sqf

```diff
@@ -0,0 +1,118 @@
+/* ----------------------------------------------------------------------------
+Function: fza_ihadss_fnc_routeCycle
+
+Description:
+    Cycle to the next route under the correct conditioning
+
+Parameters:
+    _heli - The apache
+
+Returns:
+    Nothing
+
+Examples:
+    --- Code
+    [_heli] call fza_ihadss_fnc_routeCycle
+    ---
+
+Author:
+    Snow(Dryden)
+---------------------------------------------------------------------------- */
+#include "\fza_ah64_dms\headers\constants.h"
+params ["_heli"];
+
+private _routeData    = _heli getVariable "fza_ah64_routeData";
+private _routeCurrent = _heli getVariable "fza_ah64_routeSelected";
+private _routeInfo    = _routeData # _routeCurrent;
+private _rteIndex     = _heli getVariable "fza_ah64_routeCurPnt";
+private _Course1      = -1;
+private _Course2      = -1;
+
+if (_rteIndex == -1) exitwith {};
+
+//create Route without A/B/B/C side/side duplicates
+private _rteCycleList = [];
+private _rteCycleIndex = _rteIndex;
+{
+    private _previousIndex = (_foreachindex - 1);
+    private _previousPoint = (_routeInfo#_previousIndex);
+    if (_previousPoint isEqualTo _x && _previousIndex > -1) then { 
+        if (_previousIndex < _rteIndex) then {
+            _rteCycleIndex = _rteCycleIndex - 1;
+        };
+        continue;
+    };
+    _rteCycleList pushBack _X;
+} foreach _routeInfo;
+
+private _previousPnt = if (count _rteCycleList > 1 && _rteCycleIndex > 0) then {_rteCycleList#(_rteCycleIndex - 1);} else {-1;};
+private _currentPnt  = if (count _rteCycleList > 0 && _rteCycleIndex > -1) then {_rteCycleList#_rteCycleIndex;} else {-1;};
+private _nextPnt     = if (count _rteCycleList > (_rteCycleIndex + 1)) then {_rteCycleList#(_rteCycleIndex + 1);} else {-1;};
+_previousPnt;
+private _previousPntPos = [_heli, _previousPnt, 1] call fza_dms_fnc_pointGetValue;
+private _currentPntPos  = [_heli, _currentPnt, 1] call fza_dms_fnc_pointGetValue;
+private _nextPntPos     = [_heli, _nextPnt, 1] call fza_dms_fnc_pointGetValue;
+
+if (_previousPnt isNotEqualTo -1 && _currentPnt isNotEqualTo -1 && _previousPntPos isNotEqualTo _currentPntPos) then {
+    _Course1 = _previousPntPos getDir _currentPntPos;
+};
+if (_currentPnt isNotEqualTo -1 && _nextPnt isNotEqualTo -1 && _currentPntPos isNotEqualTo _nextPntPos) then {
+    _Course2 = _currentPntPos getDir _nextPntPos;
+};
+
+private _angle = ((_Course1 + _Course2 + 180) mod 360) / 2;
+private _angle = if (_Course1 == -1 || _Course2 == -1) then {90;} else {_angle;};
+
+if (_Course2 isEqualTo -1) then {//end point
+    _angle = (_Course1 + _angle + 360) mod 360;
+};
+if (_Course1 isEqualTo -1) then {//start point
```

Comment:
Could you write what you expect this code to do? I would like to check the code against the intended algorithm